src/Sushi.Mediakiwi.API/Controllers/AssetController.cs

```diff
@@ -0,0 +1,271 @@
+                if (request.Data?.Length == 0 || string.IsNullOrWhiteSpace(request.Data?.FileName) == true)
+                {
+                    return BadRequest();
+                }
+
+                // But no gallery supplied
+                if (request.GalleryID.GetValueOrDefault(0) == 0)
+                {
+                    return BadRequest();
+                }
+            }
+
+            // The target asset
+            Data.Asset targetAsset = new Data.Asset();
+
+            // We're updating an asset, get it from DB
+            if (request.ID.GetValueOrDefault(0) > 0)
+            {
+                targetAsset = await Data.Asset.SelectOneAsync(request.ID.Value);
+            }
+
+            // We have new data provided, process it
+            if (request.Data != null)
+            {
+                targetAsset.Type = request.Data.ContentType;
+                targetAsset.Extention = request.Data.FileName.Substring(request.Data.FileName.LastIndexOf('.'));
+                targetAsset.FileName = request.Data.FileName;
+                targetAsset.Size = request.Data.Length;
+
+                // set the new filename
+                targetAsset.FileName = request.Data.FileName;
+
+                // Whenever it's an image, extract image data
+                if (request.Data.ContentType.Equals("image/jpeg", System.StringComparison.InvariantCultureIgnoreCase) ||
+                   request.Data.ContentType.Equals("image/jpg", System.StringComparison.InvariantCultureIgnoreCase) ||
+                   request.Data.ContentType.Equals("image/gif", System.StringComparison.InvariantCultureIgnoreCase) ||
+                   request.Data.ContentType.Equals("image/png", System.StringComparison.InvariantCultureIgnoreCase))
+                {
+                    targetAsset.IsImage = true; 
+                    var extension = request.Data.FileName.Substring(request.Data.FileName.LastIndexOf('.') + 1);
+                    var name = request.Data.FileName.Substring(0, request.Data.FileName.LastIndexOf('.'));
+
+                    // Create an image based on the supplied data
+                    using (var image = System.Drawing.Image.FromStream(request.Data.OpenReadStream()))
+                    {
+                        targetAsset.Width = image.Width;
+                        targetAsset.Height = image.Height;
+                        
+                        // Create a thumbnail ?
+                        if (WimServerConfiguration.Instance.Thumbnails.CreateThumbnails)
+                        {
+                            try
+                            {
+
+                                using (var thumb = CreateThumbnailImage(image))
+                                {
+                                    if (thumb != null)
+                                    {
+                                  
+                                        var thumbName = $"{name}_thumb.{extension}";
```

Comment:
Is there always an extension?

Reply:
yes, there should be, will throw a BadRequest when it's left out